HttpDownloader / Resource Download

FIr3baL opened the pull request:
DownloadProgress:
- return the initial name of requested download instead of resource.destination
DownloadFinished:
- return the initial name of requested download instead of nothing
- same change enables SpringDownloader to remove this resource-download

Solves these 2 issues:
https://github.com/beyond-all-reason/spring-launcher/issues/22
https://github.com/beyond-all-reason/spring-launcher/issues/23

FIr3baL committed:
HttpDownloader / Resource Download

DownloadProgress:
- return the initial name of requested download instead of resource.destination
DownloadFinished:
- return the initial name of requested download instead nothing
- same change enables SpringDownloader to remove this resource-download

Solves these 2 issues:
https://github.com/beyond-all-reason/spring-launcher/issues/22
https://github.com/beyond-all-reason/spring-launcher/issues/23

p2004a commented on src/spring_downloader.js:
```diff
@@ -105,9 +105,9 @@ class SpringDownloader extends EventEmitter {
 	}
 
-	downloadResource(resource) {
+	downloadResource(name, resource) {
```
This function is also called from `src/launcher_wizard.js` and you aren't fixing it there. In addition, you won't be able to call it from there correctly because resources in the config at the moment don't have names: https://github.com/beyond-all-reason/BYAR-Chobby/blob/master/dist_cfg/config.json#L21-L33.

Also below it could invoke `PrdDownloader.downloadResource` so you would have to change type signature there too as it doesn't have the name as argument at the mmoment.

Maybe instead of taking the name as argument, name should be just property of the resource that defaults to destination in emitter?

FIr3baL replied:
Ok, i just spent some time to understand the origin of configs and their resulting resource downloads...

Instead of adding a name property to every config, which doesn't add new information, I'll try to use destination as name.

p2004a commented on src/http_downloader.js:
```diff
@@ -52,34 +52,36 @@ class HttpDownloader extends EventEmitter {
 		const url = new URL(resource['url']);
-		const name = resource['destination'];
-		const destination = path.join(springPlatform.writePath, name);
+		this.name = name;
```
Why change it from local variable to object property?

FIr3baL replied:
I'll revert that.
Think i did that because name was duplicate as parametername and local name.
But 'this' correspondends to HttpDownloader in this context, not the download itself. So it's misleading.

p2004a commented on src/http_downloader.js:
```diff
@@ -52,34 +52,36 @@ class HttpDownloader extends EventEmitter {
+		const destination = path.join(springPlatform.writePath, resource['destination']);
 		if (fs.existsSync(destination)) {
-			this.emit('finished', `Skipping ${destination}: already exists.`);
+			// this.emit('finished', `Skipping ${destination}: already exists.`);
```
Plase don't keep old version of the code as comment here and in other locations. Old version is visible in git history and such comment doesn't add any value to reader.

FIr3baL replied:
Alright, i'll remove it.

FIr3baL committed:
extractor.js
- added name to parameters of emit 'finished'

http_downloader.js
- changed 'finished' and 'progress' emits to use name as second argument

FIr3baL committed:
Merge branch 'master' of https://github.com/FIr3baL/spring-launcher

# Conflicts: using mine
#	src/http_downloader.js

FIr3baL committed:
prior version
